sources/api/netdog/src/wicked/mod.rs

```diff
@@ -86,6 +86,46 @@ pub(crate) struct WickedInterface {
     pub(crate) link: Option<WickedLinkConfig>,
 }
 
+#[derive(Debug, Serialize, PartialEq)]
+pub(crate) struct WickedName {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    namespace: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "$value")]
+    name_body: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "$unflatten=permanent-address")]
+    permanent_address: Option<MacAddress>,
+}
+
+impl WickedName {
+    pub(crate) fn new(id: InterfaceId) -> Self {
+        match id {
+            InterfaceId::Name(name) => Self {
+                namespace: None,
+                name_body: Some(name.to_string()),
+                permanent_address: None,
+            },
+            InterfaceId::MacAddress(mac) => Self {
+                namespace: Some("ethernet".to_string()),
+                name_body: None,
+                permanent_address: Some(mac),
+            },
+        }
+    }
+}
```

Comment:
Nice! Its probably worth a comment for those not certain what the resulting xml must look like for wicked. I would have a hard time understanding what this is doing unless I had looked very carefully at wicked xml examples.